### ISSUE TITLE
Regression: case splitting uses variable name variants instead of the unused original names

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2729]
```agda
data Size : Set where
  ↑ : Size → Size

data Nat : Size → Set where
  zero : ∀ i → Nat (↑ i)
  suc  : ∀ i → Nat i → Nat (↑ i)

pred : ∀ i → Nat i → Nat i
pred i x = {!x!}  -- C-c C-c

-- WRONG (agda-2.5.3):
-- pred .(↑ i₁) (zero i₁) = ?
-- pred .(↑ i₁) (suc i₁ x) = ?

-- EXPECTED (correct in agda-2.5.1.1):
-- pred .(↑ i) (zero i) = ?
-- pred .(↑ i) (suc i x) = ?
```
This is a rather annoying regression; it affects interactive proof development quite a bit.

**Comment by andreasabel** [comment]
@UlfNorell regression introduced by fix of #2018 https://github.com/agda/agda/commit/67d50a9159147f0aa990aef4ae3ccbbfdd2b56df ? 

My patch https://github.com/agda/agda/commit/e60c544ed9f2234c7a2021adfae356811025d4bc bears also responsibility.

**Comment by andreasabel** [comment]
https://github.com/agda/agda/commit/67d50a9159147f0aa990aef4ae3ccbbfdd2b56df introduced fresh name picking in AbstractToConcrete, treating pattern variables in a left-to-right order.  Instead, it should do two passes: first bind the user-written patterns without freshening, then the system-inserted ones with freshening.  Then my patch https://github.com/agda/agda/commit/e60c544ed9f2234c7a2021adfae356811025d4bc should be obsolete.
